src/librustdoc/clean/types.rs

```diff
@@ -648,12 +647,38 @@ impl<'a> Iterator for ListAttributesIter<'a> {
 crate trait AttributesExt {
     /// Finds an attribute as List and returns the list of attributes nested inside.
     fn lists(&self, name: Symbol) -> ListAttributesIter<'_>;
+    
+    fn span(&self) -> Option<rustc_span::Span>;
+    
+    fn inner_docs(&self) -> bool;
+    
+    fn other_attrs(&self) -> Vec<ast::Attribute>;
 }
 
 impl AttributesExt for [ast::Attribute] {
     fn lists(&self, name: Symbol) -> ListAttributesIter<'_> {
         ListAttributesIter { attrs: self.iter(), current_list: Vec::new().into_iter(), name }
     }
+    
+    fn span(&self) -> Option<rustc_span::Span> {
```

Comment:
Wow, I'm just now realizing this is the span of the *doc string*, not of any other part of the attribute. Could you add that as a comment?
```suggestion
    /// Return the span of the first doc-comment, if it exists.
    fn span(&self) -> Option<rustc_span::Span> {
```